### DISCUSSION TITLE
Programming boards - newbie here

### DISCUSSION THREAD

**Discussion by patrykz103** [azzajess/USB-Shifter-to-Thrustmaster-Wheelbase#13] [Q&A]
Hello all,

First of all, I have zero experience about programming this type of boards. I just stopped thinking after "Connection to PC before flashing" section in https://github.com/azzajess/USB-Shifter-to-Thrustmaster-Wheelbase/blob/main/USBShield.md
Can someone explain to me what to do next in "newbie" style ? What to connect to what, which code to add in Arduino IDA... 

I ordered this USB to TTL:
![grhewerhgrtehd](https://github.com/user-attachments/assets/fe0410e6-97b4-4751-8273-7cb3f324c487)
This let me program board, yes ?
How to connect this to boards ? 
How to connect this to pc ? Just plug USB ? And how to program boards ?

I ordered same 6+r+ sequential shifter.

Many questions, zero idea how to do this 😂
You are my "last resort", i wanted to program this shifter as new device with rpi pico, but there is even more to work. 
I have Thrustmaster TX Leather Edition.

**Comment by azzajess** [comment]
Watch this video which shows you how to connect your Arduino to the TTL and how to install the drivers for it. https://www.youtube.com/watch?v=-Of5Ov2Gl4Y
**NOTE:** Just be aware that by default the programmer is going to program at 5v, however you are programing a 3v3 Arduino. So you will need to switch the voltage selector from 5v to 3v3 on the programmer that you bought. You can change it by pulling it off and connecting it to the 3.3v side.
![image](https://github.com/user-attachments/assets/063b33c7-442a-4109-ab5d-72e9e0642c40)

You can follow the video guide and program a blink example to confirm its working.

Before we go any further, I have to ask which option are you using to connect to the shifter? Are you planning to use the USB Shield route and have bought a USB Shield? or are you planning to hardwire the arduino to the shifter?

**Reply by patrykz103** [reply]
I wanted to make 2 in 1 shifter, so ordered USB Shield.
I saw couple hours ago this video, thats good explanation for me how to do this.
But now, when i want to program mini pro i need to File >open this file?:
https://github.com/azzajess/USB-Shifter-to-Thrustmaster-Wheelbase/blob/main/USBShifter2PS2/USBShifter2PS2.ino
![Screenshot_20240713_020927_YouTube](https://github.com/user-attachments/assets/f8880f8e-3b8a-4703-bb9c-4d9c8290f839)

USB Shield is only for connection to the shifter ? Or I need to program this too ?

**Reply by azzajess** [reply]
Yeah you can open the arduino file from the github for flashing. however before you do that you need to ensure you have the USB Hosts shield 2.0 library installed. 
![LibraryManager_1](https://github.com/user-attachments/assets/46d24ca5-6823-40cc-ac9a-19ac8dab948d)
Following the image above, search and import USB Hosts Shield 2.

**Reply by patrykz103** [reply]
While flashing mini pro, usb shield need to be connected to arduino ?

**Comment by azzajess** [comment]
No, ideally just without for now

**Reply by patrykz103** [reply]
Okay, I installed USB Shield library. When i open this sketch file (USBShifter2PS2.ino) do I need to add this library to USBShifter2PS2.ino file and then upload ? Or just instalation and thats all ?

**Comment by azzajess** [comment]
Just installation. It's already added to the Arduino file. Give it a flash
and see how you go. Don't forget to ensure the correct board is used in the
Arduino settings like the video

**Reply by patrykz103** [reply]
Okay, thank you very much for help. When all parts come to me, i will do all this things.
I had 8 gear shifter, but i unscrew bottom plate and saw USB-C connection, so i returned this, and order this one;
https://www.aliexpress.com/item/1005002669507696.html?spm=a2g0o.order_list.order_list_main.32.40f91802wTGWtP
Same as yours, 6 gears + reverse + extrernal (with switch), and I have one more question for today. If i program arduino with this .ino file, sequential gears will be automaticly changed from normal gears when i push this switch ? Or need to change something in this file ?

**Comment by azzajess** [comment]
Yes, nothing you need to do with the code. The only time would be if it
doesn't work due to different controller etc

**Comment by patrykz103** [comment]
Im just thinking from yesterday, if there is the way to change this sequential switch to probably handbrake (i have Thrustmaster TX leather edition) to A button ? Or this shifter PS2 connector cant handle buttons from wheel ?

**Comment by azzajess** [comment]
The Arduino is currently emulating a Thrustmaster th8a and there isn't a
handbrake on this so there isn't a code for it.
However there is that Thrustmaster handbrake add-on and I don't know what
the codes are for it to emulate. So unfortunately no

**Reply by patrykz103** [reply]
Okay, i programmed arduino, soldered all together, but dont want to work. In my opinion my shifter is not correct one, can you just look at that and tell me if its not good ?
![20240723_105748](https://github.com/user-attachments/assets/10d49e77-5002-4ad2-a446-4bd8fab450a9)
It has type of hall sensors, not switches inside.

**Reply by azzajess** [reply]
Whoa this is crazy cool. Yeah its very different to mine. 

We can try and figure out if we can manually change the code to suit the hall effect style switch?

**Reply by patrykz103** [reply]
I hope so 😁 If we cant fix this to work, i need to return this shifter, and buy th8s. So what I need to check ? Remember - I am lower than newbie in arduino 🤣 If you want to check something by me, you need to show me this in screenshot 🤣

**Reply by azzajess** [reply]
Ill try where I can. 
So you will need to reflash the arduino with modified firmware in order to debug the shifter.
https://github.com/azzajess/USB-Shifter-to-Thrustmaster-Wheelbase/blob/main/USBShield.md#flashing--debug
The link above talks a little bit about what code to uncomment in order to begin the debug process. Let me know if you get stuck at a part and need clarification. It hopefully should be enough to get some data out. When you get some data of the different codes, i can then walk you through implementing them in the code.

**Comment by patrykz103** [comment]
![wfggewrfwegr](https://github.com/user-attachments/assets/c9452237-a254-4f9f-9e37-9fec6a46ac7c)
I see when i connect this directly to PC
